app/src/main/java/com/android/sample/model/profile/ProfilesRepositoryFirestore.kt

```diff
@@ -92,4 +93,47 @@ open class ProfilesRepositoryFirestore(private val db: FirebaseFirestore) : Prof
         .addOnSuccessListener { onSuccess() }
         .addOnFailureListener { exception -> onFailure(exception) }
   }
+
+  override fun updateProfile(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+    performFirestoreOperation(
+        db.collection("profiles").document(user.id).set(user), onSuccess, onFailure)
+  }
+
+  override fun deleteActivityFromProfile(
+      userId: String,
+      activityId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection("profiles")
+        .document(userId)
+        .update("activities", FieldValue.arrayRemove(activityId))
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
+            onSuccess()
+          } else {
+            task.exception?.let { e ->
+              Log.e("UserProfileRepository", "Error deleting activity from profile", e)
+              onFailure(e)
+            }
+          }
+        }
+  }
+
+  private fun performFirestoreOperation(
+      task: Task<Void>,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    task.addOnCompleteListener { result ->
+      if (result.isSuccessful) {
+        onSuccess()
+      } else {
+        result.exception?.let { e ->
+          Log.e("TodosRepositoryFirestore", "Error performing Firestore operation", e)
+          onFailure(e)
+        }
+      }
+    }
+  }
```

Comment:
why did you create this function ? I do not think it is a good idea, as it generalises the log message, which would have been useful to have personnalised, as in line 116.

Plus, if you created this function why don't you use it in deleteActivityFromProfile function ?

Reply:
I decided to remove the helper for the moment, because it allows to put a specific log in each function, enabling us to know which operation failed. Using a helper, meant having the same log for all operations.